egs/madcat_ar/v1/local/create_line_image_from_page_image.py

```diff
@@ -0,0 +1,300 @@
+                    help='Path to the downloaded (and extracted) mdacat data file 2')
+parser.add_argument('database_path3', type=str,
+                    help='Path to the downloaded (and extracted) mdacat data file 3')
+parser.add_argument('data_splits', type=str,
+                    help='Path to file that contains the train/test/dev split information')
+args = parser.parse_args()
+
+bounding_box = namedtuple('bounding_box', ('area',
+                                         'length_parallel',
+                                         'length_orthogonal',
+                                         'rectangle_center',
+                                         'unit_vector',
+                                         'unit_vector_angle',
+                                         'corner_points'))
+
+
+def unit_vector(pt0, pt1):
+    dis_0_to_1 = sqrt((pt0[0] - pt1[0])**2 + (pt0[1] - pt1[1])**2)
+    return (pt1[0] - pt0[0]) / dis_0_to_1, \
+           (pt1[1] - pt0[1]) / dis_0_to_1
+
+
+def orthogonal_vector(vector):
+    return -1 * vector[1], vector[0]
+
+
+def bounding_area(index, hull):
+    unit_vector_p = unit_vector(hull[index], hull[index+1])
+    unit_vector_o = orthogonal_vector(unit_vector_p)
+
+    dis_p = tuple(np.dot(unit_vector_p, pt) for pt in hull)
+    dis_o = tuple(np.dot(unit_vector_o, pt) for pt in hull)
+
+    min_p = min(dis_p)
+    min_o = min(dis_o)
+    len_p = max(dis_p) - min_p
+    len_o = max(dis_o) - min_o
+
+    return {'area': len_p * len_o,
+            'length_parallel': len_p,
+            'length_orthogonal': len_o,
+            'rectangle_center': (min_p + len_p / 2, min_o + len_o / 2),
+            'unit_vector': unit_vector_p,
+            }
+
+
+def to_xy_coordinates(unit_vector_angle, point):
+    angle_orthogonal = unit_vector_angle + pi / 2
+    return point[0] * cos(unit_vector_angle) + point[1] * cos(angle_orthogonal), \
+           point[0] * sin(unit_vector_angle) + point[1] * sin(angle_orthogonal)
+
+
+def rotate_points(center_of_rotation, angle, points):
+    rot_points = []
+    ang = []
+    for pt in points:
+        diff = tuple([pt[d] - center_of_rotation[d] for d in range(2)])
+        diff_angle = atan2(diff[1], diff[0]) + angle
+        ang.append(diff_angle)
+        diff_length = sqrt(sum([d**2 for d in diff]))
+        rot_points.append((center_of_rotation[0] + diff_length * cos(diff_angle),
+                           center_of_rotation[1] + diff_length * sin(diff_angle)))
+
+    return rot_points
+
+
+def rectangle_corners(rectangle):
+    corner_points = []
+    for i1 in (.5, -.5):
+        for i2 in (i1, -1 * i1):
+            corner_points.append((rectangle['rectangle_center'][0] + i1 * rectangle['length_parallel'],
+                            rectangle['rectangle_center'][1] + i2 * rectangle['length_orthogonal']))
+
+    return rotate_points(rectangle['rectangle_center'], rectangle['unit_vector_angle'], corner_points)
+
+
+def minimum_bounding_box(points):
```

Comment:
All these functions need more documentation.  You need to document the format of the inputs and outputs and what the function does, using a doc-string-- don't rely on the user having to read the code.